src/react/components/pages/editorPage/canvas.tsx

```diff
@@ -205,26 +205,13 @@ export default class Canvas extends React.Component<ICanvasProps, ICanvasState>
         this.onWindowResize();
     }
 
-    private removeAllRegions = () => {
-        const ids = this.state.currentAsset.regions.map((r) => r.id);
-        for (const id of ids) {
-            this.editor.RM.deleteRegionById(id);
-        }
-        this.deleteRegionsFromAsset(this.state.currentAsset.regions);
-    }
-
-    private addRegions = (regions: IRegion[]) => {
-        this.addRegionsToCanvasTools(regions);
-        this.addRegionsToAsset(regions);
-    }
-
-    private addRegionsToAsset = (regions: IRegion[]) => {
+    public addRegionsToAsset = (regions: IRegion[]) => {
```

Comment:
Instead of introducing new public methods to the canvas we should just expose `regions` as a new prop that appends the regions onto the canvas if they don't already exist.